OpenRA.Mods.D2k/Traits/SpiceBloom.cs

```diff
@@ -78,6 +84,12 @@ public SpiceBloom(ActorInitializer init, SpiceBloomInfo info)
 
 		public void Tick(Actor self)
 		{
+			if (!self.World.Map.Contains(self.Location))
+				return;
+
+			if (info.GrowthTerrainTypes.Any() && info.GrowthTerrainTypes.Contains(self.World.Map.GetTerrainInfo(self.Location).Type))
```

Comment:
I think this should be `if (any && !contains)`, right?
We want to exit early if the current terrain type isn't in the set, but only if the set isn't empty.